src/App.tsx

```diff
@@ -24,22 +22,25 @@ export interface IState {
     eventTracker: EventTracker;
 }
 
-export default class App extends React.Component <IProps, IState> {
+export default class App extends React.Component<IProps, IState> {
     private name: string;
+    private cManager: ChoicesManager;
+    private eManager: EventsManager;
 
     constructor(props: IProps) {
         super(props);
 
         const playerStats = new PlayerStats();
         const eventTracker = new EventTracker(
-            [new BoomerGregorEvent(), new CuteGirlEvent(), new FratPartyEvent()],
-            [new LandingEvent(), new PickFacultyEvent()]
+            [events.BoomerGregorEvent, events.CuteGirlEvent, events.FratPartyEvent],
+            [events.LandingEvent, events.PickFacultyEvent, events.PickResidenceEvent]
         );
         let firstEvent = eventTracker.getNextEvent();
 
         //TODO: should we auto allocate a dummy player name or allow user to input their own?
         this.name = "P1";
-
+        this.cManager = new ChoicesManager(choices);
+        this.eManager = new EventsManager(events);
```

Comment:
This is the nittiest nit pick, but it might be easier to read `eventManager` and `choiceManager` instead of just `eManager` and `cManager`.